shared/lib/multichain/chain-agnostic-permission-utils/caip-accounts.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  Caip25CaveatValue,
+  InternalScopesObject,
+  parseScopeString,
+} from '@metamask/chain-agnostic-permission';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import { CaipAccountId, parseCaipAccountId } from '@metamask/utils';
+
+/**
+ * Gets all accounts from an array of scopes objects
+ * This extracts all account IDs from both required and optional scopes
+ * and returns a unique set.
+ *
+ * @param scopesObjects - The scopes objects to extract accounts from
+ * @returns Array of unique account IDs
+ */
+export function getCaipAccountIdsFromScopesObjects(
+  scopesObjects: InternalScopesObject[],
+): CaipAccountId[] {
+  const allAccounts = new Set<CaipAccountId>();
+
+  for (const scopeObject of scopesObjects) {
+    for (const { accounts } of Object.values(scopeObject)) {
+      for (const account of accounts) {
+        allAccounts.add(account);
+      }
+    }
+  }
+
+  return Array.from(allAccounts);
+}
```

Comment:
Previously was:
```
export function getAllAccountsFromScopesObjects(
  scopesObjects: InternalScopesObject[],
): CaipAccountId[] {
  return Array.from(
    new Set([...scopesObjects.flatMap(getAllAccountsFromScopesObject)]),
  );
}
```

Reply:
The main thing that needed to be removed was the two extra uniques, which this does